src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/network-request-failures.mdx

```diff
@@ -0,0 +1,959 @@
+        )
+```
+
+## Description [#description]
+
+The New Relic Android SDK API provides several methods to [track network requests](/docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/noticehttptransaction-android-agent-api). If a network request fails, you can record details about the failure with `noticeNetworkFailure`. In most cases, place this call inside exception handlers, such as catch blocks.
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+
+      <th width={200}>
+        Type
+      </th>
+      
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$url`
+      </td>
+
+      <td>
+       `string`
+      </td>
+
+      <td>
+        Required. The URL of the request.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$httpMethod`
+      </td>
+
+     <td>
+        `string`
+      </td>
+
+      <td>
+        Required. The HTTP method used, such as GET or POST.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$startTime`
+      </td>
+
+        <td>
+        `long`
+      </td>
+
+      <td>
+        Required. The start time of the request in milliseconds since the epoch.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$endTime`
+      </td>
+
+      <td>
+        `long`
+      </td>
+      <td>
+        Required. The end time of the request in milliseconds since the epoch.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$exception`
+      </td>
+
+      <td>
+        `string`
+      </td>
+      <td>
+        Either this or `$failure` parameter is required. This is the exception that occurred. New Relic can automatically translate many common exceptions into network failure types.
+      </td>
+    </tr>
+
+    <tr>
+      <td>
+        `$failure`
+      </td>
+
+      <td>
+        `enum`
+      </td>
+
+      <td>
+        Either this or `$exception` parameter is required. The type of network failure that occurred. If an exception cannot be resolved to a network failure automatically, this method can be used to categorize the failure accurately. The values are defined by the `NetworkFailure` enum. Valid values include `Unknown`, `BadURL`, `TimedOut`, `CannotConnectToHost`, `DNSLookupFailed`, `BadServerResponse`, and `SecureConnectionFailed`.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Examples [#examples]
+
+Here’s an example of an error listener that uses an error as part of the noticed network failure to New Relic:
+
+### Java [#java] 
+
+```java
+new Response.ErrorListener() {
+                            @Override
+                            public void onErrorResponse(Error error) {
+                                NewRelic.noticeNetworkFailure(badUrl, "GET", System.nanoTime(), System.nanoTime(), NetworkFailure.exceptionToNetworkFailure(error));
+                            }
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
NewRelic.noticeNetworkFailure(badUrl, "GET", System.nanoTime(), System.nanoTime(), NetworkFailure.exceptionToNetworkFailure(error))

Reply:
```suggestion
```kotlin

    NewRelic.noticeNetworkFailure(badUrl, "GET", System.nanoTime(), System.nanoTime(), NetworkFailure.exceptionToNetworkFailure(error))

```
```